src/app/routes/assets/asset-enrolment-list/asset-enrolment-list.component.ts

```diff
@@ -51,6 +51,18 @@ export class AssetEnrolmentListComponent implements OnInit, OnDestroy {
   }
 
   back() {
-    this.location.back();
+    this.urlService.previous.pipe(
+      takeUntil(this.subscription$)
+    ).subscribe(url => {
+      this.navigateBack(url);
+    });
+  }
+
+  navigateBack(url: string): void {
```

Comment:
Should `navigateBack()` be private? If not, could the naming be improved? The names don't imply a difference in the two functions, to me at least.

Reply:
Yes, it should be private. Maybe also rename it into `navigateBackHandler` ?

Reply:
Sounds good